### PR TITLE
Remove legacy collator related code

### LINKED ISSUE #8241: Remove legacy collator code and remove icu dependency
## Enhancement
Clickhouse's collator support depends on ICU lib, while TiFlash implement its own support for collator that is compactible with TiDB/MySQL, and it is not depend on ICU lib. This issue aims at removing the legacy Clickhouse's collator releated code and removing the dependency of ICU lib

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #8241

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by flowbehappy — APPROVED** [review]

File: dbms/src/Core/SortDescription.cpp [hunk 1]
```diff
@@ -26,8 +26,7 @@ std::string SortColumnDescription::getID() const
 {
     WriteBufferFromOwnString out;
     out << column_name << ", " << column_number << ", " << direction << ", " << nulls_direction;
-    if (collator)
-        out << ", collation locale: " << collator->getLocale();
+    out << ", collation locale: " << collator->getCollatorId();
```
Comment (JaySon-Huang) [inline dbms/src/Core/SortDescription.cpp:29]: I think we should keep the nullptr check?

Comment (windtalker) [reply dbms/src/Core/SortDescription.cpp:29]: Yes :(

Comment (SeaRise) [inline dbms/src/Core/SortDescription.cpp]: Why delete `if (collator)`?

**Comment by zanmato1984** [comment]
https://github.com/pingcap/tiflash/blob/90cde744f0cfc825619300c1624e14fbd266eedf/dbms/src/Common/config.h.in#L5 This needs to be removed as well.

**Review by SeaRise — APPROVED** [review]
lgtm

**Comment by windtalker** [comment]
> https://github.com/pingcap/tiflash/blob/90cde744f0cfc825619300c1624e14fbd266eedf/dbms/src/Common/config.h.in#L5
> 
> 
> and
> https://github.com/pingcap/tiflash/blob/90cde744f0cfc825619300c1624e14fbd266eedf/dbms/src/Common/config_build.cpp.in#L56C1-L57
> These lines need to be removed as well.

Done

**Review by zanmato1984 — APPROVED** [review]
LGTM

**Pull request merged**